kalite/distributed/static/js/distributed/user/views.js

```diff
@@ -314,4 +314,64 @@ window.UserView = BaseView.extend({
     logout: function() {
         this.model.logout();
     }
-});
\ No newline at end of file
+});
+
+/* This view toggles which navbar items are displayed to each type of user */ 
+window.ToggleNavbarView = BaseView.extend ({
+
+    template: HB.template("user/navigation"),
+
+    initialize: function() {
+
+        _.bindAll(this);        
+        this.listenTo(this.model, "change", this.render);
+        $("topnav").append(this.template());
+
+    },
+
+    render: function() {
+        
+        this.facility_id = {"facility_id": this.model.get("facility_id")};
+        this.$el.html(this.template(this.facility_id));
+        
+        this.userView = new UserView({ model: this.model, el: "#topnav" });
+        $("#topnav").prepend(new AutoCompleteView().$el);
+        
+        // GUEST
+        if ( !this.model.get("is_logged_in") ) {
+            this.$(".guest").removeClass("display-none");
+        }
+        // ALL LOGGED IN
+        if ( this.model.get("is_logged_in") ) {
+            this.$(".logged-in").removeClass("display-none");
+        }
+        // ALL USERS - slight lag in tabs appearing, seems unnecessary but ensures that all tabs appear at the same time
+        if ( this.model.get("is_logged_in") || !this.model.get("is_logged_in") ) {
+            this.$(".learn-tab").removeClass("display-none");
+        }
+        // STUDENTS
+        if ( this.model.get("is_logged_in") && !this.model.get("is_admin") ) {
+            this.$(".student").removeClass("display-none");
+        }
+        // TEACHER
+        if ( !this.model.get("is_django_user") && this.model.get("is_admin") ) {
+            this.$(".teacher").removeClass("display-none");
+        }
+        // ADMINS
+        if ( this.model.get("is_django_user") ) {
+            this.$(".admin").removeClass("display-none");
+        }
+
+        // ACTIVATES NAV TAB ACCORDING TO CURRENT PAGE
```

Comment:
This seems like a decent approach.

Instead of using ==, though, you could just check that the start of the URL contains the relevant url:

```
pageType: function() {
if ( window.location.pathname.search("/coachreports/table/") > -1 ) {
return "teachPage";
} 
if ( window.location.pathname.search("/learn/khan/") > -1) {
return "learnPage";
} 
if ( window.location.pathname.search("/management/zone/None/") > -1 ) {
return "managePage";
}
```

However, as you rightly suggest, this hardcoding doesn't work for the management page, (it is also vulnerable to URL changes on the server side). Instead you could use the newly added Urls object to generate the URL dynamically:

```
pageType: function() {
if ( window.location.pathname.search(Urls.coach_reports()) > -1 ) {
return "teachPage";
} 
if ( window.location.pathname.search(Urls.learn()) > -1) {
return "learnPage";
} 
if ( window.location.pathname.search(Urls.zone_redirect()) > -1 ) {
return "managePage";
}
```